### ISSUE TITLE
Is there a whichwrap option?

### ISSUE THREAD

**Issue by irevoire** [helix-editor/helix#768]
Hello,

I would love to have a [whichwrap](http://vimdoc.sourceforge.net/htmldoc/options.html#%27whichwrap%27) option like in vim.
Basically, the idea would be that, when I'm at the end of a line and type `right`, I should just stay at the end of my line.
The same goes for the start of the line and `left`.

Is this something planned?

**Comment by kirawi** [comment]
From the documentation, it sounds like the `Home` (next line) and `End` (prev line) keys. https://docs.helix-editor.com/keymap.html

**Comment by archseer** [comment]
@kirawi you should read the linked documentation, it's about configuring this behavior, not about specific keys :P

**Comment by archseer** [comment]
There's no configuration option and I'm not sure if we want this to be configurable. 

Currently everything wraps: h/l used to stop at the end of line but after a discussion with @cessen we removed this limitation and had it match kakoune. I personally don't use this wrapping behavior, but what's the rationale for explicitly disabling it?

**Comment by irevoire** [comment]
I would love to see this behaviour back, it was one of the reason why I went back to vim after using kakoune for ~6 months ☹️

(But overall I'm really in love with helix and don't plan to quit soon even if this behaviour is never implemented! Thanks for all the great works!)

**Comment by Omnikar** [comment]
I personally think this can be useful, if a bit niche, but I don't really think there's much of a reason *not* to make this configurable.

**Comment by yowayb** [comment]
@irevoire just curious what the benefit would be?

**Comment by ArijanJ** [comment]
> curious what the benefit would be?

Just preventing complexity, IMO.

We've always had 4 directional keys, and making half of them go diagonally only at certain times is plain confusing.

**Comment by irevoire** [comment]
Yeah, especially when you have multiple cursors and by hitting `right` one goes to the next line :weary:

**Comment by hholst80** [comment]
@pantos9000 is your code something that is useful yet for the general public, or more specifically, is it something I dare to test without having my dreams and expectations shattered? :)

**Comment by pantos9000** [comment]
@hholst80 Yup, this is ready and should work as expected. Just did a small test on my side to confirm. Well, at least it works as expected by me :stuck_out_tongue_closed_eyes: 

Note that my PR #10576 introduces two different kinds of movement sets: `same_line` (solving this issue) and `anchored` movement. If you want, I can also split up the PR - I kept the logic for both in separate commits, so having two separate PRs won't be a big deal for me.
